File: 민석/그리디 알고리즘/1026-보물.py
n = int(input())

a = list(map(int, input().split()))
b = list(map(int, input().split()))

a.sort(reverse=False)
b.sort(reverse=True)

k = 0

for i in range(n):
  k = k + int(a[i]) * int(b[i])

print(k)

File: 민재/그리디/1026_보물.swift
//
//  보물.swift
//  algo
//
//  Created by 김민재 on 2022/05/13.
//

import Foundation


let n = Int(readLine()!)!

var A = readLine()!.split(separator: " ").map {Int($0)!}
var B = readLine()!.split(separator: " ").map {Int($0)!}


A.sort(by: >)
B.sort(by: <)

var answer = 0
for i in 0..<n{
    answer += A[i] * B[i]
}

print(answer)

